src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -283,5 +283,24 @@ public override Uri GetUri(string name)
         {
             return new Uri(_directory.Uri, name);
         }
+
+        public override async Task<bool> IsUpdateRequiredAsync(Uri resourceUri, string id, string version)
+        {
+            //the destination blob
+            var blobName = resourceUri.Segments.Last();
+            var blobRelativePath = $"{id}/{version}/{blobName}";
```

Comment:
This is leaking knowledge of the destination URL from `DnxMaker`. The URL create should be shared.

Reply:
Updated